addons/account/wizard/account_merge_wizard.py

```diff
@@ -0,0 +1,310 @@
+
+        if account.deprecated:
+            other_name_elements.append(_("Deprecated"))
+
+        if not self.is_group_by_name:
+            grouping_key_name = account_type_label
+            if other_name_elements:
+                grouping_key_name = f'{grouping_key_name} ({", ".join(other_name_elements)})'
+        else:
+            grouping_key_name = f'{account.name} ({", ".join([account_type_label] + other_name_elements)})'
+
+        return grouping_key_name
+
+    @api.depends('is_group_by_name', 'account_ids')
+    def _compute_wizard_line_ids(self):
+        """ Determine which accounts to merge together. """
+        for (i, wizard) in enumerate(self):
+            # Filter out Bank / Cash accounts
+            accounts = wizard.account_ids._origin.filtered(lambda a: a.account_type not in ('asset_bank', 'asset_cash'))
+
+            wizard_lines_vals_list = []
+            sequence = 0
+            for grouping_key, group_accounts in accounts.grouped(wizard._get_grouping_key).items():
+                grouping_key_str = str((i, *grouping_key))  # The grouping key identifies both the wizard and the account properties
+                wizard_lines_vals_list.append({
+                    'display_type': 'line_section',
+                    'info': wizard._get_group_name(group_accounts[0]),
+                    'grouping_key': grouping_key_str,
+                    'sequence': (sequence := sequence + 1),
+                })
+                for account in group_accounts:
+                    wizard_lines_vals_list.append({
+                        'display_type': 'account',
+                        'account_id': account.id,
+                        'grouping_key': grouping_key_str,
+                        'is_selected': True,
+                        'sequence': (sequence := sequence + 1),
+                    })
+
+            wizard.wizard_line_ids = [
+                Command.unlink(wizard_line.id)
+                for wizard_line in wizard.wizard_line_ids
+            ] + [
+                Command.create(vals)
+                for vals in wizard_lines_vals_list
+            ]
+
+    @api.depends('wizard_line_ids.is_selected', 'wizard_line_ids.info')
+    def _compute_disable_merge_button(self):
+        for wizard in self:
+            wizard_lines_to_merge = wizard.wizard_line_ids.filtered(lambda l: l.display_type == 'account' and l.is_selected and not l.info)
+            wizard.disable_merge_button = all(
+                len(wizard_line_group) < 2
+                for wizard_line_group in wizard_lines_to_merge.grouped('grouping_key').values()
+            )
+
+    def action_merge(self):
+        """ Merge each group of accounts in `self.wizard_line_ids`. """
+        self._check_access_rights(self.account_ids)
+
+        for wizard in self:
+            wizard_lines_selected = wizard.wizard_line_ids.filtered(lambda l: l.display_type == 'account' and l.is_selected and not l.info)
```

Comment:
Since `info` is stored, there *could* be a concurrency issue where it shouldn't be merge-able between the moment when the wizard was opened and when the merge action was called.

Very unlikely and I don't think it would be a huge issue if that happened.
The most plausible issue would be that an entry became hashed in the meantime.

Would it change something performance-wise to not store `info`?

Reply:
It would change practically nothing as `info` is really quick to recompute. I'll make it non-stored. Same for `account_has_hashed_entries`.